### PR TITLE
#419 Admin paginated Invitation table view PR

### PR DESCRIPTION
**Pull request by Ndevu12**
### PR Description

This PR empower ADMIN with the capability of retrieving invitations by email, and also ADMIN can retrieve all invitations and visualize them in a paginated table.

### Description of tasks that were expected to be completed

- [x] Admin should retrieve invitation by anything
- [x] Define both get invitation by anything resolver and its schema
- [x] Admin should be able to retreive all invitations and visualize them in a paginated table
- [x] Define get all invitation resolver and its schema
- [x] Add pagination feature to both, get all invitation resolver and its schema

### Functionality

Both,  get invitation by email resolver and   get all invitation resolver check if a user is logged in before doing anyhing else. When a user is not logged in, they return GrapgQL error, when a user is logged in and is not ADMIN or MANAGER it return a GraphQL error. When user is ADMIN, they perform the expected operations.

### How has this been tested?

It is very simple To test this functionality.

Here are few steps on how You can test it.

Run these commands in order.

```javascript
git clone https://github.com/atlp-rwanda/atlp-pulse-bn.git
```

```javascript
cd ft-invitation-table
npm install
```

```javascript
npm run dev
```

Write this in your browser to open apollo graphql studio:

```javascript
https://studio.apollographql.com/sandbox/explorer
```

On apollo graphql studio, Create queries which are the same as the following but you must make sure that you have Invitations in you data base.

N.B: Make sure you have created MongoDB connection before. 

Then here the querries:

Get a invitations query:

```javascript
query AllInvitations {
  getAllInvitations(limit: 10, offset: 0) {
    invitations {
      invitees {
        email
        role
      }
    }
    totalInvitations
    }
  }
}
```

Get invitation by email:

```javascript
query Invitation{
  getInvitations(query: "ntwarichar@gmail.com") {
invitations {
    invitees {
      email
      role
    }
    status
    id
}
  }
}
```

### PR Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code generate no warnings
- [x] My test coverage meet the set test coverage threshold
- [x] There are no vulnerabilities
- [x] There are no conflicts with the base branch

### CODE REVIEW

**Review by Sophie20202 — APPROVED** [review]

**Review by UwicyezaG — APPROVED** [review]

**Review by niyobertin — APPROVED** [review]

**Review by jniyonkuru — APPROVED** [review]

**Comment by jniyonkuru** [comment]
yet you have 2 commits,squash them

**Review by ncutixavier — COMMENTED** [review]
I can see a lot of unnecessary changes, could you fix that

**Comment by Ndevu12** [comment]
> I can see a lot of unnecessary changes, could you fix that

SOLVED "All unnecessary changes have been resolved" Thanks

File: src/resolvers/team.resolvers.ts
Comment (ncutixavier) [inline src/resolvers/team.resolvers.ts:1]: What did you change in this file?

File: yarn.lock
Comment (ncutixavier) [inline yarn.lock:1]: We are using npm not yarn, so `yarn.lock` is not needed

**Comment by ncutixavier** [comment]
@Ndevu12 , @GSinseswa721 , @niyobertin check above comments

**Comment by Ndevu12** [comment]
> @Ndevu12 , @GSinseswa721 , @niyobertin check above comments

All have been resolved

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**